src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2000,6 +1945,9 @@ void MergeTreeData::waitForOutdatedPartsToBeLoaded() const TSA_NO_THREAD_SAFETY_
     if (isStaticStorage())
         return;
 
+    /// We need to load parts as fast as possible
+    getOutdatedPartsLoadingThreadPool().enableTurboMode();
```

Comment:
Now it is safe, but a little bit fragile. IMO having something like `ScopedTurboMode turbo_mode(getOutdatedPartsLoadingThreadPool());` would be really nice.

Reply:
We can use SCOPE_EXIT macros.